### PR TITLE
WFCORE-1374 - make code work with jdk9b112 and newer

### PR DESCRIPTION
**Pull request by ctomc**
- upgrade jboss-modules
- WFCORE-1290 upgrade jboss-marshalling

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -105,10 +105,10 @@
         <version.org.jboss.logging.jboss-logging-tools>2.0.1.Final</version.org.jboss.logging.jboss-logging-tools>
         <version.org.jboss.logmanager.jboss-logmanager>2.0.3.Final</version.org.jboss.logmanager.jboss-logmanager>
         <version.org.jboss.logmanager.log4j-jboss-logmanager>1.1.2.Final</version.org.jboss.logmanager.log4j-jboss-logmanager>
-        <version.org.jboss.marshalling.jboss-marshalling>1.4.10.Final</version.org.jboss.marshalling.jboss-marshalling>
+        <version.org.jboss.marshalling.jboss-marshalling>1.4.11.Final</version.org.jboss.marshalling.jboss-marshalling>
```
Comment (rhusar) [inline pom.xml]: Component upgrade Jira https://issues.jboss.org/browse/WFCORE-1290

File: testsuite/domain/pom.xml [hunk 2]
```diff
@@ -177,7 +177,6 @@
                     <argLine>${jvm.args.ip.client} ${jvm.args.timeouts} ${surefire.system.args}</argLine>
 
                     <systemPropertyVariables>
-                        <jboss.options>-ea ${surefire.system.args}</jboss.options>
```
Comment (bstansberry) [inline testsuite/domain/pom.xml]: Why?

How is -ea getting set on the PC and HC process without this?

What other values were there in ${surefire.system.args} and how are they getting set on the PC and HC process without this?

File: testsuite/domain/pom.xml [hunk 3]
```diff
@@ -240,7 +239,6 @@
                             <argLine>${jvm.args.ip.client} ${jvm.args.timeouts} ${surefire.system.args}</argLine>
 
                             <systemPropertyVariables>
-                                <jboss.options>${surefire.system.args}</jboss.options>
```
Comment (bstansberry) [inline testsuite/domain/pom.xml]: Same question as above.

File: testsuite/domain/pom.xml (hunk 2 above)
Comment (ctomc) [inline testsuite/domain/pom.xml]: This is somewhat an oversight, only -ea should have been removed as it is by default now present in surefire.system.args already. parameter itself should stay.

Comment (ctomc) [inline testsuite/domain/pom.xml]: fixed

File: testsuite/domain/pom.xml (hunk 3 above)
Comment (ctomc) [inline testsuite/domain/pom.xml]: fixed

File: testsuite/domain/pom.xml (hunk 2 above)
Comment (ctomc) [inline testsuite/domain/pom.xml]: Also fixed on #1609

Comment (bstansberry) [inline testsuite/domain/pom.xml]: testsuite/pom.xml overrides the setting from pom.xml and does not include -ea.

**Pull request merged**